core/src/types/operator/operator_futures.rs

```diff
@@ -215,6 +215,57 @@ impl Future for FutureAppender {
     }
 }
 
+/// Future that generated by [`Operator::presign_read_with`].
+///
+/// Users can add more options by public functions provided by this struct.
+pub struct FuturePresignRead(pub(crate) OperatorFuture<(OpRead, OpPresign), PresignedRequest>);
```

Comment:
`OpPresign` should be built by `OpRead`, so we need `(OpRead, Duration)` here.

Reply:
Done.